### PR TITLE
Introduces 'focusable' in FocusNode and FocusScopeNode, deprecating 'canRequestFocus' for focus handling

### PR DESCRIPTION
**Pull request by Mairramer**
This PR adds the focusable parameter to deprecate/replace the canRequestFocus mapped in https://github.com/flutter/flutter/issues/24722 and described in https://github.com/flutter/flutter/issues/154148

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

### CODE REVIEW

**Comment by Piinks** [comment]
> This PR adds the focusable parameter to replace the canRequestFocus mapped in https://github.com/flutter/flutter/issues/24722

@Mairramer thanks for the PR! That issue is specifically tracking changes we won't make because it would be too breaking. I am not sure if we would want to make this change for that reason.

File: packages/flutter/lib/src/material/ink_well.dart [hunk 1]
```diff
@@ -662,7 +674,7 @@ class InkResponse extends StatelessWidget {
       enableFeedback: enableFeedback,
       excludeFromSemantics: excludeFromSemantics,
       focusNode: focusNode,
-      canRequestFocus: canRequestFocus,
+      focusable: focusable && canRequestFocus,
```
Comment (Piinks) [inline packages/flutter/lib/src/material/ink_well.dart:677]: What if one is false and the other is true?

**Comment by Piinks** [comment]
>> This PR adds the focusable parameter to replace the canRequestFocus mapped in https://github.com/flutter/flutter/issues/24722
>
> That issue is specifically tracking changes we won't make because it would be too breaking.

@Mairramer I do not think we will be merging this change based on the above feedback. Do you have any reason why we must break the API in this way?

File: packages/flutter/lib/src/material/ink_well.dart (hunk 1 above)
Comment (Mairramer) [reply packages/flutter/lib/src/material/ink_well.dart:677]: What I figured is that if either case is `false`, the result will be `false`. This simplifies the four possible scenarios and prevents anyone using it from being surprised. Also, considering that the intention is to completely replace and not use both options.

Comment (Piinks) [reply packages/flutter/lib/src/material/ink_well.dart:677]: The feedback about why we need to make this change has not been addressed yet. Without a reason, we will likely close this PR.

**Comment by Piinks** [comment]
Hey @Mairramer for the reasons previously stated, I am going to close this PR. It os not worth breaking the existing API, and we consistently use canRequestFocus throughout the framework.

**Pull request closed without merging**